### ISSUE TITLE
`TestDefinition.name` should be optional.

### ISSUE THREAD

**Issue by jespertheend** [denoland/deno#13195]
If it's possible to do
```js
Deno.test(() => {});
```
wouldn't it make sense to also be able to omit the `name` property like so?
```js
Deno.test({
  fn: () => {},
});
```

For reference: 
- https://doc.deno.land/deno/stable/~/Deno.test
- https://doc.deno.land/deno/stable/~/Deno.TestDefinition

**Comment by kitsonk** [comment]
Why would this be desirable?

**Comment by bartlomieju** [comment]
@nayeemrmn suggested it in https://github.com/denoland/deno/pull/12749#issuecomment-968052317 and I kinda forgot about it. I think it might be useful, though it's definitely on the "nice-to-have" spectrum, not "required-to-be-productive".

**Comment by kitsonk** [comment]
@nayeemrmn suggested anonymous functions, not making the name optional and had a good justification for them. Why would omitting the name be a nice to have?

**Comment by bartlomieju** [comment]
@kitsonk just to be clear - I believe that "definition" overload should still require either an explicit name or named function, but I think it's fine for "shorthand" form not to require it (as Nayeem suggested). But I'm not in favor of having "anonymous" as a name and would prefer to have a name derived from location of the test definition.

**Comment by jespertheend** [comment]
Ah sorry, I wasn't aware the "shorthand" signature actually doesn't work right now.

After looking at the documentation I noticed the shorthand signature, so I used that since my file only had a single test. At some point I needed extra permissions, so I decided to use to use the "definition" overload. To my surprise I was getting type errors for omitting the `name` field now.

That's what prompted me to create this issue, I was mostly surprised about the incompatibility between the two signatures. I simply thought it was a bug. But if I had known the shorthand signature doesn't work yet, or even that it would use `anonymous` as name, I wouldn't have used it in the first place.
